lib/iris/_data_manager.py

```diff
@@ -185,7 +199,15 @@ def _assert_axioms(self):
                 'real data and realised {!r}.')
         assert state, emsg.format(self._realised_dtype)
 
-    def _deepcopy(self, memo, data=None, realised_dtype='none'):
+        state = not (self.has_lazy_data() and
+                     self._lazy_array.dtype.kind != 'f' and
+                     self._realised_dtype is not None)
```

Comment:
Note that,
* the `realised_dtype` will only be set if it is different from the managed data dtype.
* the `realised_dtype` can only be of the kind `biu`
* for the masked 'biu' case that's lazified, it will be represented as a float dask array
* therefore, only a float dask array may have a `realised_dtype`

Of course, it's completely possible to create a float dask, and assign it along with a `realised_dtype` that's `biu`, as a naughty way to cast the final result ... that that case is really hard to catch - we can live with it as a corner-case and it'll melt away when dask rolls out masked support.